docs/typescript.md

```diff
@@ -0,0 +1,9 @@
+This project uses [TypeScript](https://en.wikipedia.org/wiki/TypeScript).
+
+The fact that different parts of this project are built for different target architectures, namely browsers and node servers, is reflected in the TypeScript configuration. The different targets are organized in individual files. Notable differences are the `src/` files included and libraries available.
+
+* `tsconfig.base.json` The base configuration that is inherited by all other configurations
+* `tsconfig.server.json` The configuration used to build for the server
+* `tsconfig.json` The configuration used to build for the client
+    > This file should really be called `tsconfig.client.json` but this is currently not possible as not all tools (which one?) allow for the tsconfig name to be configurable
```

Comment:
@bitPogo @jakobw Which?

Reply:
the vue-cli typescript plugin is the issue, I believe.
https://cli.vuejs.org/config/#typescript
https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli-plugin-typescript/index.js#L35

Reply:
Thanks